systemtest/src/test/java/io/strimzi/systemtest/BaseST.java

```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
         }
     }
 
+    private Properties stringToProperties(String str) {
+        Properties result = new Properties();
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(str);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private Properties configMap2Properties(ConfigMap cm) {
+        return stringToProperties(cm.getData().get("server.config"));
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+        return validLines;
+    }
+
+    protected void checkKafkaConfiguration(String podNamePrefix, Map<String, Object> config, String clusterName) {
+        LOGGER.info("Checking kafka configuration");
+        List<Pod> pods = kubeClient().listPodsByPrefixInName(podNamePrefix);
+
+        Properties properties = configMap2Properties(kubeClient().getConfigMap(clusterName + "-kafka-config"));
+
+        config.forEach((key, val) -> {
```

Comment:
I think use here classic for is more readable.

You can loop over config items and some this: `assertThat(properties, hasEntry(configEntry));`